File: Java_Backend/E-Tour/src/main/java/entity/Cost.java
package entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Cost {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer costId;
    
    private Double singlePersonCost;
    private Double extraPersonCost;
    private Double childWithBed;
    private Double childWithoutBed;
    private LocalDate validFrom;
    private LocalDate validTo;
    
    @ManyToOne
    @JoinColumn(name = "package_id")
    private TourPackage tourPackage;
}

File: DotNet_Backend/SafarnamaApplication/SafarnamaApplication/Models/Cost.cs
using System;
using System.Collections.Generic;

namespace SafarnamaApplication.Models;

public partial class Cost
{
    public int CostId { get; set; }

    public int? PackageId { get; set; }

    public double? SinglePersonCost { get; set; }

    public double? ExtraPersonCost { get; set; }

    public double? ChildWithBed { get; set; }

    public double? ChildWithoutBed { get; set; }

    public DateOnly? ValidFrom { get; set; }

    public DateOnly? ValidTo { get; set; }

    public virtual Package? Package { get; set; }
}
